File: Gfg/C/firstindex.c
//Given an array of length N and an integer x, you need to find and return the first index of integer x present in the array. Return -1 if it is not present in the array.
//Test Cases
//Input: Size:5 Array:4 5 6 7 4 Element:4
//Output: Index:0

//Input: Size:6 Array:1 2 3 3 4 3 Element:3
//Output: Index:2

#include<stdio.h>
#include<stdlib.h>
int first_index(int arr[],int size,int x,int k){
    if(k==size-1){
        return -1;
    }
   k++;
   if(arr[k]==x)
   {
       return k;
   }
   return first_index(arr,size,x,k);
}
int main(){
int N,x,i,index,k;
printf("Enter the size of array:");
scanf("%d",&N);
int arr[N];
printf("Enter the elements of array:");
for(i=0;i<N;i++)
  scanf("%d",&arr[i]);
printf("Enter the element for which you need the first index:");
scanf("%d",&x);
k=-1;
index = first_index(arr,N,x,k);
printf("Index is: %d",index);
return 0;

}

File: Gfg/First_index.java
import java.util.*;
import java.lang.*;
public class First_index {

    static int first_index(int[] a,int i,int s,int y){
        if(s==0) // if size of array is 0
            return -1;
        if(a[i]==y && i<s)
            return i;
        else{
           int v= first_index(a,i+1,s,y); //recursive function
           if(v==-1)
               return -1;
           else
              return v;
    }
    }
    public static void main(String []args){

        System.out.println("enter the size of array");
        Scanner sc=new Scanner(System.in);
        int n;
        n=sc.nextInt();
        int[] arr=new int[n];
        for(int i=0;i<n;i++) {
        arr[i]=sc.nextInt();
        }
        System.out.println("enter the element to be searched");
        int x;
        x=sc.nextInt();
        int index=first_index(arr,0,n,x);
        System.out.println(index);
    }
}

/*
INPUT
6
4
3
2
2
1
5
2
OUTPUT 2

 */